### DISCUSSION TITLE
How to passing char* arr[] to JavaScript in EM_JS?

### DISCUSSION THREAD

**Discussion by iliakan** [emscripten-core/emscripten#18881] [Q&A]
Here's my (incorrect) attempt to implement `execve`:

```c
EM_JS(int, execve, (const char *pathname, char *const argv[], char *const envp[]), {
  return Module.execve(pathname, argv, envp);
});
```

It doesn't work, because the generated JS code in module file looks like this:
```js
function execve(pathname,argv[],envp[]) { return Module.execve(pathname, argv, envp); }
```

(The problem is that `[]` in function arguments cause syntax error in JS)

How should I implement it correctly? 

P.S. I tried to use `char **const argv, char **const envp` as arguments (instead of `*...[]`), but then the signature doesn't match, and the compilation fails.

**Comment by sbc100** [comment]
We should probably be stripping those off yes.   You should be able to two `*` instead though.   e.g.:

```
EM_JS(int, execve, (const char *pathname, char ** argv, char ** envp), {
  return Module.execve(pathname, argv, envp);
});
```

However if you just want to call into en external JS function you might want to consider using `--js-library` file, and skip the whole EM_JS macro.
